src/services/jsDoc.ts

```diff
@@ -256,18 +260,19 @@ namespace ts.JsDoc {
     function getCommentOwnerInfo(tokenAtPos: Node): CommentOwnerInfo | undefined {
         // TODO: add support for:
         // - enums/enum members
-        // - interfaces
         // - property declarations
         // - potentially property assignments
         for (let commentOwner = tokenAtPos; commentOwner; commentOwner = commentOwner.parent) {
             switch (commentOwner.kind) {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.MethodDeclaration:
                 case SyntaxKind.Constructor:
-                    const { parameters } = commentOwner as FunctionDeclaration | MethodDeclaration | ConstructorDeclaration;
+                case SyntaxKind.MethodSignature:
+                    const { parameters } = commentOwner as FunctionDeclaration | MethodDeclaration | ConstructorDeclaration | MethodSignature;
                     return { commentOwner, parameters };
 
                 case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.InterfaceDeclaration:

```

Comment:
Looks like `commentOwner` is only used in one place: `if (commentOwner.getStart() < position) { return undefined; }`.
Seems like this function could just be `getParameters`...

Reply:
I like this current setup since we have an explicit list of what places are permissible for these templates, and it's easy to add new things.

Reply:
If a user types `/**` in an empty file and hits enter, I would expect to see an empty doc comment -- it's unambiguous that that's what they're trying to do. No sense in not giving the user formatting just because they're not above a declaration.

Reply:
Even if they're in the middle of an expression? For example, what if someone's adding a comment to a parameter (`foo(/*param*/ undefined)`) and then accidentally hit * twice. I guess they can just hit undo but it seems easy to trigger.

Reply:
Do we fill in the comment immediately on the second `*` or do we wait for them to hit enter?

Reply:
It's immediate.

Reply:
In vscode you must hit enter. What if a user wants to add a single-line comment to an interface?

Reply:
I mean same as my description above: if they accidentally hit a second star, then it will trigger, but otherwise it won't. Seems somehow less bad then triggering in the middle of a parenthetical block.

Reply:
I mean a single-line jsdoc comment. These seem pretty common.
```ts
/** My interface */
interface I {
    /** My method */
    m(): void;
}
```

Reply:
Huh. Yeah we never use that with the current implementation.

Reply:
So maybe we should just autocomplete to empty, one-line jsDoc comments and then if there are params, autocomplete to a multi-line and fill in the params?